test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -141,49 +168,94 @@ public void testObjLitAssignmentDepth4() {
   public void testGlobalObjectDeclaredToPreserveItsPreviousValue1() {
     test("var a = a ? a : {}; a.c = 1;",
          "var a = a ? a : {}; var a$c = 1;");
+
+    test("var a = a ? a : {}; /** @nocollapse */ a.c = 1;",
+        "var a = a ? a : {}; a.c = 1;");
   }
 
   public void testGlobalObjectDeclaredToPreserveItsPreviousValue2() {
     test("var a = a || {}; a.c = 1;",
          "var a = a || {}; var a$c = 1;");
-  }
+
+    test("var a = a || {}; /** @nocollapse */ a.c = 1;",
+        "var a = a || {}; a.c = 1;");
+ }
 
   public void testGlobalObjectDeclaredToPreserveItsPreviousValue3() {
     test("var a = a || {get b() {}}; a.c = 1;",
          "var a = a || {get b() {}}; var a$c = 1;");
+
+    test("var a = a || {get b() {}}; /** @nocollapse */ a.c = 1;",
+        "var a = a || {get b() {}}; a.c = 1;");
```

Comment:
I don't see any difference between the before/after here, except that the nocollapse annotation is removed. But in other tests you are using testSame. Why is the annotation removed in some tests and not in others? Or can you just use testSame here?